plugins/ssr-plugin.js

```diff
@@ -92,18 +93,61 @@ const SSRBodyTemplate = createPlugin({
         }
       }
 
-      const criticalChunkScripts = getLoaderScript(ctx, {
-        legacyUrls,
-        modernUrls,
-      });
+      let preloadHints = [];
+      let criticalChunkScripts = [];
+
+      const browser = ctx.useragent.browser;
+
+      /*
+      Edge must get transpiled classes due to:
+      - https://github.com/Microsoft/ChakraCore/issues/5030
+      - https://github.com/Microsoft/ChakraCore/issues/4663
+      - https://github.com/babel/babel/issues/8019
+      Rather than transpile classes in the modern bundles, Edge should be forced on the slow path
+      */
+      if (browser.name !== 'Edge') {
+        for (let url of modernUrls) {
+          const crossoriginAttr = url.startsWith(__webpack_public_path__)
+            ? ''
+            : ' crossorigin="anonymous"';
+          preloadHints.push(
+            `<link rel="modulepreload" href="${url}"${crossoriginAttr}/>`
+          );
+          criticalChunkScripts.push(
+            `<script type="module" defer src="${url}" nonce="${
+              ctx.nonce
+            }"${crossoriginAttr}></script>`
+          );
+        }
+      }
+
+      const isSafari10_1 =
+        (browser.name === 'Safari' || browser.name === 'Mobile Safari') &&
+        browser.version.startsWith('10.1');
+
+      // Safari 10.1 supports modules but not `nomodule` attribute.
+      if (!isSafari10_1) {
```

Comment:
Might be reading this wrong but it looks like non-edge, non-safari 10.1 browsers get both modern and legacy bundles in `criticalChunkScripts`

Reply:
Yep, that's exactly as intended.

Browsers that do not support `<script type="module">` will simply ignore them. And all browser that support `<script type="module">` will ignore `<script nomodule>` scripts, *except* for Safari 10.1, which will download and execute both `type="module"` *and* `nomodule`. So exclusively for Safari 10.1, only `<script type="module">` should be sent.

Reply:
👍